src/Symfony/Component/OptionsResolver/OptionsResolver.php

```diff
@@ -965,6 +954,42 @@ public function offsetGet($option)
     }
 
     /**
+     * Verify value is of the allowed type. Recursive method to support
+     * typed array notation like ClassName[], or scalar arrays (int[]).
+     *
+     * @param string $type  the required allowedType string
+     * @param mixed  $value the value
+     *
+     * @return bool Whether or not $value if of the allowed type
+     */
+    private function verifyAllowedType($type, $value)
+    {
+        $valid = false;
+
+        $type = isset(self::$typeAliases[$type]) ? self::$typeAliases[$type] : $type;
+
+        if (substr($type, -2) === '[]') {
+            if (is_array($value)) {
```

Comment:
It is very hard to read what happens when `if (!is_array())`.

I would omit the single return and add a `if (!is_array()) {return false;}`